src/main/java/com/users/service/UserService.java

```diff
@@ -0,0 +1,78 @@
+package com.users.service;
+
+import com.users.dto.UserDto;
+import com.users.entities.User;
+import com.users.exceptions.EntityAlreadyExistsException;
+import com.users.exceptions.EntityNotFoundException;
+import com.users.mappers.UserMapper;
+import com.users.repository.UserRepository;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Transactional
```

Comment:
pentru ce folosesti adnotarea asta

Reply:
la ce te referi cand zici mai multe operatinui, si daca ai un exemplu

Reply:
din intelegerea mea este ca de ex daca am o metoda de update in care eu vreau sa schimb mai multe atribute si ajung a unul cum e dateOfBirth si eu pun sub 18 ani si mi arunca o eroare, nu mi ia nici celelalte atribute pe care vreau sa le schimb. Sau de ex daca in alta metoda incerc sa fac mai multe schimbari la ce am in baza de data si este aruncata o eroare pe undeva, nu o sa imi ia nici o schimbare pana in mom respectiv.

Reply:
nop, un exemplu simplu e o metoda de transfer de bani, prima data iti verifica contul tau daca ai suma pe care vrei sa o transferi, iti ia banii din cont, si dupa ii transfera in alt cont si face acolo alte check uri, gen daca contul ala exista etc, daca operatia de a extrage bani de la tine din cont a mers bine, dar mutarea lor in celalalt cont pica, prima operatiune se face roll back, adica banii tai revin in cont iar starea bazei de date nu e periclitata.

Operatiune se refera la request uri catre baza de date, daca tu faci un singur request in acea metoda catre baza de date nu e obigatoriu sa pui Transactional

Un lucru bun e ca pe metodele care doar fac get la baza de date (gen nu faci un update, insterg sau un delete in metoda ta) sa pui adnotare la nivel de metoda @Transactional(readOnly = true), poti sa citesti un pic de ce e recomandat asa